tasks/main.yml

```
@@ -106,6 +107,38 @@
   tags:
     - wg-config
 
+- name: Create private key for unmanaged hosts
+  shell: "wg genkey | tee {{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  args:
+    creates: "{{ wireguard_remote_directory }}/{{ item.host }}-privatekey"
+  register: uh_privkey
+  with_items: "{{ wireguard_unmanaged_hosts | default([]) }}"
```

```suggestion
  with_items: "{{ wireguard_unmanaged_hosts }}"
```
If you put everything in a `block` as suggested above then there is no need for a default value as the block is only executed if `wireguard_unmanaged_hosts` is defined.